devcontainer-setup.md

```diff
@@ -0,0 +1,226 @@
+## Core Configuration Files
+
+*   **`.devcontainer/devcontainer.json`**: The primary configuration file. It tells the GitHub Actions how to create and
+    access the dev container which will run your code
+*   **`Dockerfile`** (optional): If you need an advanced container configuration
+*   **`docker-compose.yml`** (optionally): A YAML file for defining and running multi-container Docker applications,
+    can be combined with `Dockerfile` for advanced configuration of containers
+
+---
+
+## Scenario 1: Plain `devcontainer.json` (Using a Pre-built Image)
+
+This is the simplest approach, suitable when a pre-built Docker image on a registry (like Docker Hub) meets your needs.
+
+**When to use:**
+*   You need just a standard environment (e.g., `mcr.microsoft.com/vscode/devcontainers/java`).
+*   No custom software installation or complex setup is required beyond
+    what the image and devcontainer features and extensions provide
+
+
+**File Structure:**
+```
+your-project/
+└── .devcontainer/
+    └── devcontainer.json
+```
+
+**`.devcontainer/devcontainer.json` Example:**
+```json
+{
+  "name": "Java",
+  "image": "mcr.microsoft.com/devcontainers/java:1-21",
+  "features": {
+    "ghcr.io/devcontainers/features/java:1": {
+      "version": "none",
+      "installMaven": "true",
+      "mavenVersion": "3.8.6"
+    }
+  }
+}
+```
+**Explanation:**
+* Github Actions pulls the specified `image` and runs it.
+* Your local project folder is mounted into `/workspace` (by default, unless you specify `workspaceFolder` property) inside the container.
+* `features` and `customizations` are applied to the container (additional features installed, etc.)
+
+**Sample repository:**
+
+https://github.com/jetbrains-junie/2048
```

Comment:
- Check the formatting.
- Rename repositories and ensure all repositories are public.

Additionally, it would be great to have an example where we have JVM repositories, and Junie fixes something and runs the tests.